samu.c

```diff
@@ -90,6 +91,50 @@ jobsflag(const char *flag)
 	buildopts.maxjobs = num > 0 ? num : -1;
 }
 
+static void
+jobserverflags(const char *flag)
+{
+	int rfd, wfd;
+
+	if (!flag)
+		return;
+	if (sscanf(flag, "%d,%d", &rfd, &wfd) == 2) {
+		/* prepare error message */
+		errno = EBADF;
+	} else if (strncmp(flag, "fifo:", 5) == 0) {
+		rfd = open(flag + 5, O_RDONLY);
+		wfd = open(flag + 5, O_WRONLY);
```

Comment:
You should just open it once as `O_RDWR` and set both `rfd` and `wfd` to the same value.

Also, we should print a message if this happens.

Reply:
Print a message whenever a FIFO is used?